src/keyfile.c

```diff
@@ -99,3 +112,29 @@ void setStringValueNum(Key_File *key_file, size_t num, void *v) {
   size_t length = strlen(value) + 1;
   snprintf(key_file->file_entry[num].value = malloc(length), length, value);
 }
+
+void setBoolValueNum(Key_File *kf, size_t num, void *v) {
+  char *value = (char*) v;
+  if (!*value) { errno = EINVAL; return; }
+
+  size_t length, hash;
+  char *tmp;
+
+  length = strlen(value);
+  snprintf(tmp = malloc(length + 1), length + 1, value);
+
+  hash = hashstring(toLowerCase(tmp));
+
+  if ((*value == '1' && length == 1) || hash == _yes_ || hash == _true_) {
+    kf->file_entry[num].value = realloc(kf->file_entry[num].value, 5);
+    snprintf(kf->file_entry[num].value, 5, "true");
```

Comment:
This pattern is used very often: Outsource into function, replace with `free` / `strcpy`.